hub/core/chunk_engine.py

```diff
@@ -535,6 +539,14 @@ def read_bytes_for_sample(self, global_sample_index: int) -> bytes:
         sb, eb = chunk.byte_positions_encoder[local_sample_index]
         return buffer[sb:eb].tobytes()
 
+    def read_shape_for_sample(self, global_sample_index: int) -> Tuple[int, ...]:
+        enc = self.chunk_id_encoder
+        chunk = self.get_chunks_for_sample(global_sample_index)[0]
```

Comment:
could you add an if else here. If len(chunks) == 1, continue with the current logic, else read the shape from tile encoder